### PR TITLE
Docs: GitHub Pages instructions

### LINKED ISSUE #6187: docs RFE: more explicit context about when the tip: "Use the github-pages gem" applies
- [x] I believe this to be a bug, not a question about using Jekyll.
- [x] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [x] This is a feature request.

---

- [x] I was trying to install.
- [x] I was trying to build.
- [x] It was another bug.

## My Reproduction Steps

Read docs.  (quickstart stuff, followed by deployment)

https://jekyllrb.com/docs/github-pages/#deploying-jekyll-to-github-pages

Read tip: Use the github-pages gem

Become confused as to why this is considered a useful tip.

> Using it in your projects means that when you deploy your site to GitHub Pages, you will not be caught by unexpected differences between various versions of the gems.

It should be possible to avoid being caught by unexpected differences without any special action.  You use Gemfile.lock (which is created by default), and you deploy generated HTML.  How would upgrading versions on someone else's schedule _help_ handle "unexpected differences", or as the pages-gem puts it, "breaking changes"?  

## The Output I Wanted

Tips should have a coherent reason that you would want to follow the tip.

This might be addressed if the Github section was super-explicit that *none* of it is written to help you deploy HTML.  *I guess my root problem was that `github-pages.md` wasn't very helpful in deciding whether I wanted to deploy as HTML or not.*

I eventually went back and read through the tutorial it links to for people who haven't used GitHub Pages before (which I have).  Judging by this, the point is to let people edit pages using the GitHub website.

Then the purpose of the tip, is if you want to let people edit a site using both GitHub and a local checkout with Jekyll, depending on their current situation.

(Personally I'm surprised that github-pages "Ruby itself is versioned" does not merit some up-front caveats.  But I don't have any experience or current interest using it, so for all I know it's a perfected bootstrap with no more requirements than Jekyll).

It took me some time to come to this conclusion, partly because I did not already have a clear understanding of Jekyll, Gemfiles, Ruby, and Github Pages.  That's why I was reading the docs :).

No offence if this report doesn't clear the bar for constructiveness (e.g. specific wordings).  I was just logging my setup, and with lines like "don't install the github-pages gem, it doesn't do anything I want", it suggested the possibility of an improvement.

/cc @jekyll/documentation

### PR DESCRIPTION
**Pull request by DirtyF**
- [x] Clarify `pages-gem` intent (fix #6187)
- [x] Update Gemfile example,
- [x]  Use URL filters for links to assets.
- [x] Move up link to GitHub Pages Help and Support.

### CODE REVIEW

File: docs/_docs/github-pages.md [hunk 1]
```diff
@@ -5,102 +5,80 @@ permalink: /docs/github-pages/
...
+##  The github-pages gem
+
+Our friends at GitHub have provided the
+[github-pages](https://github.com/github/pages-gem) gem which is used to manage
+[Jekyll and its dependencies on GitHub Pages](https://pages.github.com/versions/).
+Using it in your projects means that when you deploy your site to GitHub Pages,
+you will not be caught by unexpected differences between various versions of the
+gems.
+
+GitHub Pages runs in `safe` mode and only allow [a set of whitelisted plugins](https://help.github.com/articles/configuring-jekyll-plugins/#default-plugins),
```
Comment (pup) [inline docs/_docs/github-pages.md]: s/allow/allows

File: docs/_docs/github-pages.md [hunk 2]
```diff
@@ -5,102 +5,80 @@ permalink: /docs/github-pages/
...
+using the `github-pages` gem instead of Jekyll provides you with the same
```
Comment (pup) [inline docs/_docs/github-pages.md]: maybe rewrite this to `, providing you with the same environment`?

File: docs/_docs/github-pages.md [hunk 3]
```diff
@@ -149,12 +127,13 @@ to see more detailed examples.
   </p>
 </div>
 
-<div class="note">
-  <h5>GitHub Pages Documentation, Help, and Support</h5>
+<div class="note info">
+  <h5>Installing <code>github-pages</code> gem on Windows</h5>
```
Comment (pup) [inline docs/_docs/github-pages.md]: `Installing the [...]`

File: docs/_docs/github-pages.md [hunk 4]
```diff
@@ -149,12 +127,13 @@ to see more detailed examples.
...
+
   <p>
-    For more information about what you can do with GitHub Pages, as well as for
-    troubleshooting guides, you should check out
-    <a href="https://help.github.com/categories/github-pages-basics/">GitHub’s Pages Help section</a>.
-    If all else fails, you should contact <a href="https://github.com/contact">GitHub Support</a>.
+    While Windows is not officially supported, it is possible
+    to install <code>github-pages</code> gem on Windows.
```
Comment (pup) [inline docs/_docs/github-pages.md]: same as above

File: docs/_docs/github-pages.md (hunk 1 above)
Comment (DirtyF) [reply docs/_docs/github-pages.md]: I wish we had [Sedy](https://marmelab.com/sedy/) on the repo 😄

File: docs/_docs/github-pages.md [hunk 5]
```diff
@@ -5,102 +5,80 @@ permalink: /docs/github-pages/
 
 [GitHub Pages](https://pages.github.com) are public web pages for users,
 organizations, and repositories, that are freely hosted on GitHub's
-`github.io` domain or on a custom domain name of your choice. GitHub Pages are
-powered by Jekyll behind the scenes, so in addition to supporting regular HTML
-content, they’re also a great way to host your Jekyll-powered website for free.
+`github.io` domain or on a custom domain name of your choice. GitHub Pages are powered by Jekyll behind the scenes, so they're a great way to host your Jekyll-powered website for free.
```
Comment (ashmaroli) [inline docs/_docs/github-pages.md]: IMO, `GitHub Pages` is a singular entity.. so the sentence should **ideally** be `GitHub Pages is powered by..`

Also, it'd be better if the line was split at around 80 chars like elsewhere on this doc..

File: docs/_docs/github-pages.md [hunk 6]
```diff
@@ -5,102 +5,80 @@ permalink: /docs/github-pages/
...
+Your site is automatically generated by GitHub Pages when you push your source
+files. Note that GitHub Pages works equally well for regular HTML content,
+simply because Jekyll treats files without YAML front matter as static assets.
+So if you only need to push generated HTML, you're good to go without any further setup.
 
 Never built a website with GitHub Pages before? [See this marvelous guide by
-Jonathan McGlone to get you up and running](http://jmcglone.com/guides/github-pages/).
-This guide will teach you what you need to know about Git, GitHub, and Jekyll to create your very own website on GitHub Pages.
+Jonathan McGlone to get you up and
+running](http://jmcglone.com/guides/github-pages/). This guide will teach you
```
Comment (ashmaroli) [inline docs/_docs/github-pages.md]: i believe the link text could be better shortened as follows:
```diff
- [See this marvelous guide by Jonathan McGlone to get you up and running](http://jmcglone.com/guides/github-pages/)
+ [See this marvelous guide by Jonathan McGlone](http://jmcglone.com/guides/github-pages/) to get you up and running
```

File: docs/_docs/github-pages.md (hunk 1 above)
Comment (ashmaroli) [reply docs/_docs/github-pages.md]: split line at around 80 chars..

File: docs/_docs/github-pages.md [hunk 7]
```diff
@@ -5,102 +5,80 @@ permalink: /docs/github-pages/
...
-but when GitHub generates your pages from the gh-pages branch all the URLs
+but when GitHub generates your pages from the `gh-pages` branch all the URLs
 will resolve properly.
 
 ## Deploying Jekyll to GitHub Pages
 
 GitHub Pages work by looking at certain branches of repositories on GitHub.
-There are two basic types available: user/organization pages and project pages.
+There are two basic types available: [user/organization and project pages](https://help.github.com/articles/user-organization-and-project-pages/).
 The way to deploy these two types of sites are nearly identical, except for a
```
Comment (ashmaroli) [inline docs/_docs/github-pages.md]: split lit at around 80 chars.. ?

**Comment by DirtyF** [comment]
Thanks for the feedback, incorporated your suggestions.

File: docs/_docs/github-pages.md (hunk 1 above)
Comment (pup) [reply docs/_docs/github-pages.md]: ^ that too

**Comment by DirtyF** [comment]
@jekyllbot: merge +docs

**Pull request merged**